Bundle/CoreBundle/Command/LegacyLinkMigratorCommand.php

```diff
@@ -0,0 +1,94 @@
+<?php
+namespace Victoire\Bundle\CoreBundle\Command;
+
+use Doctrine\ORM\Tools\DisconnectedClassMetadataFactory;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Victoire\Bundle\CoreBundle\Entity\Link;
+
+class LegacyLinkMigratorCommand extends ContainerAwareCommand
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function configure()
+    {
+        parent::configure();
+
+        $this
+            ->setName('victoire:legacy:linkMigrator')
+            ->setDescription('migrate old LinkTrait architecture into Link entity');
+    }
+
+    /**
+     * Read declared business entities and BusinessEntityPatternPages to generate their urls
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     *
+     * @return void
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $progress = $this->getHelperSet()->get('progress');
+        $progress->setProgressCharacter('V');
+        $progress->setEmptyBarCharacter('-');
+
+        $entityManager = $this->getContainer()->get('doctrine.orm.entity_manager');
+
+        $cmf = new DisconnectedClassMetadataFactory();
+        $cmf->setEntityManager($entityManager);
+        $metadatas = $cmf->getAllMetadata();
+        $classes = [];
+        $output->writeln('<info>Parse every classes to know which ones are related to Link</info>');
+        $progress->start($output, count($metadatas));
+        foreach ($metadatas as $key => $metadata) {
+            $progress->advance();
+            if ($metadata->hasAssociation('link')) {
```

Comment:
The relation name could be different from "link" (eg. in case of multiple link widget). Je propose plutôt d'utiliser la méthode "getAssociationMappings" pour récupérer les relations de l'entité courante puis de regarder si on a une relation vers Victoire\Bundle\CoreBundle\Entity\Link